v2/bigquery-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/templates/BigQueryToElasticsearch.java

```diff
@@ -0,0 +1,259 @@
+      FailsafeElementCoder.of(
+          NullableCoder.of(StringUtf8Coder.of()), NullableCoder.of(StringUtf8Coder.of()));
+
+  /** TableRow/String Coder for FailsafeElement. */
+  static final FailsafeElementCoder<TableRow, String> FAILSAFE_TABLE_ROW_ELEMENT_CODER =
+      FailsafeElementCoder.of(TableRowJsonCoder.of(), NullableCoder.of(StringUtf8Coder.of()));
+
+  /** The default suffix for error tables if dead letter table is not specified. */
+  private static final String DEFAULT_DEADLETTER_TABLE_SUFFIX = "_error_records";
+
+  /* Logger for class. */
+  private static final Logger LOG = LoggerFactory.getLogger(BigQueryToElasticsearch.class);
+
+  /**
+   * Main entry point for pipeline execution.
+   *
+   * @param args Command line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    BigQueryToElasticsearchReadOptions options =
+        PipelineOptionsFactory.fromArgs(args)
+            .withValidation()
+            .as(BigQueryToElasticsearchReadOptions.class);
+
+    run(options);
+  }
+
+  /**
+   * Runs the pipeline with the supplied options.
+   *
+   * @param options The execution parameters to the pipeline.
+   * @return The result of the pipeline execution.
+   */
+  private static PipelineResult run(BigQueryToElasticsearchReadOptions options) {
+
+    // Create the pipeline.
+    Pipeline pipeline = Pipeline.create(options);
+
+    // Register the coders for pipeline
+    CoderRegistry coderRegistry = pipeline.getCoderRegistry();
+
+    coderRegistry.registerCoderForType(
+        FAILSAFE_ELEMENT_CODER.getEncodedTypeDescriptor(), FAILSAFE_ELEMENT_CODER);
+
+    coderRegistry.registerCoderForType(
+        FAILSAFE_TABLE_ROW_ELEMENT_CODER.getEncodedTypeDescriptor(),
+        FAILSAFE_TABLE_ROW_ELEMENT_CODER);
+
+    /*
+     * Steps: 1) Read records from BigQuery via BigQueryIO.
+     *        2) Create string from Table Row.
+     *        3) Write records to Elasticsearch.
+     *
+     *
+     * Step #1: Read from BigQuery. If a query is provided then it is used to get the TableRows.
+     */
+    PCollectionTuple convertedRows =
+        pipeline
+            .apply(
+                "ReadFromBigQuery",
+                ReadBigQuery.newBuilder()
+                    .setOptions(options.as(BigQueryToElasticsearchReadOptions.class))
+                    .build())
+            /*
+             * Step #2: Convert table rows to JSON documents.
+             */
+            .apply(
+                "TableRowsToJsonDocument",
+                TableRowToJsonDocument.newBuilder()
+                    .setOptions(options.as(JavascriptTextTransformerOptions.class))
+                    .setTransformDeadletterOutTag(TRANSFORM_DEADLETTER_OUT)
+                    .setTransformOutTag(TRANSFORM_OUT)
+                    .setUdfDeadletterOutTag(UDF_TRANSFORM_DEADLETTER_OUT)
+                    .setUdfOutTag(UDF_OUT)
+                    .build());
```

Comment:
This might not be needed based on the previous comment.

Reply:
This has been removed.